### PR TITLE
PluginController: Allow enabling / disabling plugins

### LINKED ISSUE #113: Add ability to enable / disable snaps
Snaps should have a boolean state property called `enabled`. This indicates whether the snap is allowed to be started. We plan to allow users to toggle this state via snap settings in the UI, and potentially set snaps that crash repeatedly to `disabled`.

### PR DESCRIPTION
**Pull request by shanejonas**
This adds enable/disable functionality to the `PluginController`.

fixes #113

### CODE REVIEW

**Comment by shanejonas** [comment]
One thing that I wanted some feedback on is wether or not calling disable should stop the plugin

**Review by rekmarks — CHANGES_REQUESTED** [review]
In addition to inline comments, we should add a docstring for `_transitionPluginState` and `pluginStatusStateMachineConfig` to explain what they are for, how plugin statuses work and should be used, and that we are using XState conventions.

File: packages/controllers/src/plugins/PluginController.ts [hunk 1]
```diff
@@ -163,12 +164,27 @@ export enum PluginStatusEvent {
...
 
+/**
+ * Guard transitioning when the plugin is disabled.
+ */
+const disabledGuard = (serializedPlugin: SerializablePlugin) => {
+  return serializedPlugin.enabled;
+};
+
+/**
+ * The StateMachine configuration for a plugins `status` state.
```
Comment (shanejonas) [inline packages/controllers/src/plugins/PluginController.ts:175]:
```suggestion
 * The state machine configuration for a plugins `status` state.
```

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**